### PR TITLE
fix(FOROME-687): compound het

### PR DESCRIPTION
**Pull request by Lapk1n**
https://quantori.atlassian.net/browse/FOROME-687
PR into 659 branch to collect all func attrs fixes into one branch

### CODE REVIEW

File: src/pages/filter/ui/panels/function-panel/components/inheritance-mode.tsx [hunk 1]
```diff
@@ -1,201 +1,230 @@
...
-    const [variants, setVariants] = useState([])
-    const [problemGroups, setProblemGroups] = useState<string[]>([])
+  const { filterName, filterGroup, problemGroups, complexVariants } =
+    functionPanelStore
 
-    const [problemGroupValues, variantsValues] = [
-      cachedValues?.problemGroups || [],
-      cachedValues?.variants || [],
-    ]
+  const filteredVariants: [string, number][] = useMemo(() => {
```
Comment (AndreyUstyumenko) [inline src/pages/filter/ui/panels/function-panel/components/inheritance-mode.tsx]: Since you're using the same function in the other PR, it definitely should be a store getter

File: src/pages/filter/ui/panels/function-panel/components/compound-het.tsx [hunk 2]
```diff
@@ -1,112 +1,130 @@
...
   const [statFuncStatus, setStatFuncStatus] = useState<IStatFuncData>()
 
-  const initialApprox = cachedValues?.approx || options[0].value
+  const initialApprox: string =
+    cachedValues?.conditions.approx || options[0].value
 
-  const fetchStatFuncAsync = async (
-    param?: Record<string, string | string[]>,
-  ) => {
+  const handleSetConditions = (approx: string | null, variants?: string[]) => {
```
Comment (AndreyUstyumenko) [inline src/pages/filter/ui/panels/function-panel/components/compound-het.tsx:38]: It's a good practice to separate functions that are not directly related to the component. Whether move outside the component, or move to an external hook, or move to a store in order to prevent generating this function over and over each time the component updates.

File: src/pages/filter/ui/panels/function-panel/components/inheritance-mode.tsx [hunk 3]
```diff
@@ -1,201 +1,230 @@
...
-  },
-)
+  }
+
+  // to avoid displaying this data on the another func attr
+  useEffect(() => {
+    return () => filterStore.resetStatFuncData()
+  }, [])
+
+  return (
```
Comment (AndreyUstyumenko) [inline src/pages/filter/ui/panels/function-panel/components/inheritance-mode.tsx]: again, too much code in JSX. Try to split it

File: src/pages/filter/ui/panels/function-panel/function-panel.store.ts [hunk 4]
```diff
@@ -25,6 +32,23 @@ class FunctionPanelStore {
     return filterStore.selectedGroupItem.vgroup
   }
 
+  public get problemGroups(): string[] {
+    let attrData: any
```
Comment (AndreyUstyumenko) [inline src/pages/filter/ui/panels/function-panel/function-panel.store.ts:45]: are you sure it's any?

File: src/pages/filter/ui/panels/function-panel/components/compound-het.tsx [hunk 5]
```diff
@@ -1,112 +1,130 @@
...
+    functionPanelStore.setCachedValues<ICompoundHetCachedValues>(
+      FuncStepTypesEnum.GeneRegion,
+      {
+        conditions: { approx: approx, state: null },
+        variants: variants || ['Proband'],
+      },
+    )
+  }
+
+  const fetchStatFuncAsync = async () => {
```
Comment (AndreyUstyumenko) [inline src/pages/filter/ui/panels/function-panel/components/compound-het.tsx:48]: If this handler does more than just fetching it should be called differently

File: src/pages/filter/ui/panels/function-panel/components/compound-het.tsx [hunk 6]
```diff
@@ -1,112 +1,130 @@
...
     const statFuncData = await filterStore.fetchStatFuncAsync(
```
Comment (AndreyUstyumenko) [inline src/pages/filter/ui/panels/function-panel/components/compound-het.tsx:49]: Doesn't this function do the same thing as `fetchStatFunc` from `functionPanelStore`? In other places you use `fetchStatFunc`, better to stick with the same approach

File: src/pages/filter/ui/panels/function-panel/components/inheritance-mode.tsx (hunk 1 above)
Comment (Lapk1n) [reply src/pages/filter/ui/panels/function-panel/components/inheritance-mode.tsx]: fixed in https://github.com/ForomePlatform/Anfisa-React-Client/pull/444

File: src/pages/filter/ui/panels/function-panel/components/inheritance-mode.tsx (hunk 3 above)
Comment (Lapk1n) [reply src/pages/filter/ui/panels/function-panel/components/inheritance-mode.tsx]: fixed in https://github.com/ForomePlatform/Anfisa-React-Client/pull/444
